lib/libvfio-user.c

```diff
@@ -1430,6 +1430,17 @@ vfu_setup_region(vfu_ctx_t *vfu_ctx, int region_idx, size_t size,
 
     assert(vfu_ctx != NULL);
 
+    if ((flags & ~(VFU_REGION_FLAG_MASK)) ||
+        (!(flags & VFU_REGION_FLAG_RW))) {
```

Comment:
Probably not important, but why not allow a region is not readable/writeable, is not an I/O region, and there are no callabacks?

Reply:
What would the point in such a region be? nothing could ever interact with it could it?

Reply:
agreed